module/Motion/DirectionChanger.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file DirectionChanger.cpp
+ * @brief 方向転換クラス
+ * @author YKhm20020
+ */
+
+#include "DirectionChanger.h"
+
+using namespace std;
+
+DirectionChanger::DirectionChanger(COLOR _targetColor, double _targetDistance, double _targetSpeed,
+                                   int _targetAngle, bool _isClockwise, bool& _isLeftEdge,
+                                   bool _nextEdge)
+  : BlockMotion(0.88, 0.92),  // 動作時間, 失敗リスク TODO: 測定し直す
+    targetColor(_targetColor),
+    targetDistance(_targetDistance),
+    targetSpeed(_targetSpeed),
+    targetAngle(_targetAngle),
+    isClockwise(_isClockwise),
+    isLeftEdge(_isLeftEdge),
+    nextEdge(_nextEdge){};
+
+void DirectionChanger::run()
+{
+  // 事前条件を判定する
+  if(!isMetPrecondition()) {
+    return;
+  }
+
+  DistanceStraight ds(targetDistance, targetSpeed);
+  ColorStraight cs(targetColor, targetSpeed);
+
+  // 回転方向を判定
+  if(isClockwise == true) {
+    AngleRotation rotation(targetAngle, targetSpeed, isClockwise);
+    rotation.run();
+  } else {
+    AngleRotation rotation2(targetAngle, targetSpeed, isClockwise);
+    rotation2.run();
+  }
+
+  //エッジ切り替え
+  if(targetAngle == 180) {  // 後ろ方向に方向転換する場合
+    nextEdge = !isLeftEdge;
+    EdgeChanging ec(isLeftEdge, nextEdge);
+    ec.run();
+
+  } else if(isClockwise == false) {  // 左方向に方向転換する場合
+    nextEdge = false;
+    EdgeChanging ec2(isLeftEdge, nextEdge);
+    ec2.run();
+
+  } else {  // 右方向に方向転換する場合
+    nextEdge = true;
+    EdgeChanging ec3(isLeftEdge, nextEdge);
+    ec3.run();
+  }
```

Comment:
無駄なインスタンス化は減らせるなら減らしたい

```suggestion
  if(targetAngle == 180) {  // 後ろ方向に方向転換する場合
    nextEdge = !isLeftEdge;
  } else {
    nextEdge = isClockwise;
  }
  EdgeChanging ec(isLeftEdge, nextEdge);
  ec.run();
```